src/Maestro/maestro-angular/src/app/model/build-status.ts

```diff
@@ -3,13 +3,23 @@
 
 export interface BuildListResult {
   count: number;
-  value: BuildStatus[];
+  value: BuildStatusCompleted[] | BuildStatusInProgress[];
```

Comment:
The types aren't different.  It's a a BuildStatus.  The format of the object doesn't change between the two. This seems to introduce some weird types that aren't really reflective of the real state.

Reply:
I'd just add the "_links" value to the existing BuildStatus and use that.  I think "finishTime" technically needs to get a ? to make it correct, but that's not all that interesting.

Reply:
Yup, that's what I've tried to do in the first place (to add a new property to the existing interface + adding an optional ? to finishTime/result and length as they aren't in the object if the status is inProgress). But then I had some issues elsewhere in the code, things like:
![image](https://user-images.githubusercontent.com/43049559/93133076-f1e56b80-f6d6-11ea-826d-deef2e085765.png)

I didn't know what would be the correct fix to such things, that's why I added a new interface.

But we could do this:
![image](https://user-images.githubusercontent.com/43049559/93133462-8bad1880-f6d7-11ea-9f0c-801892c6415a.png)
That'd be okay?

Reply:
Yeah, that would be fine, because you know finishTime isn't null because of the previous "inProgress" checks.  There is probably some crazy way to tell TypeScript the two are related, but the ! seems accurate enough.  @alexperovich might have some TS wizardry I don't know about to link them. :-)

Reply:
There is a way to have TS do this, but you have to encode the type in a different way. They are called Discriminated Unions https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

Once the types are defined the `b.status === "inProgress"` check would narrow the type of `b`.